src/group_theory/sylow.lean

```diff
@@ -495,4 +495,79 @@ end
 
 end pointwise
 
+/-- A sylow subgroup in G is also a sylow subgroup in a subgroup of G. -/
+def subtype {p : ℕ} (P : sylow p G) (N : subgroup G) (h : ↑P ≤ N) : sylow p N :=
```

Comment:
I wonder if this should be generalized to injective monoid_homs, or even monoid_homs with p-group kernel. Then you would deduce this def as a special case.

Reply:
Hmm, probably. How would you call it? `sylow.comap`?

Reply:
> or even monoid_homs with p-group kernel

Are there more assumptions needed when generalizing to arbitrary `ϕ`? Maybe `P ≤ ϕ.range` (which is one of my `sorry`s when trying to adjust the old proof)

Reply:
The other one is
```
G: Type u
_inst_1: group G
p: ℕ
K: Type u_1
_inst_2: group K
ϕ: K →* G
hϕ: is_p_group p ↥(ϕ.ker)
Q: subgroup K
hQ: is_p_group p ↥Q
hle': P.to_subgroup ≤ Q'
⊢ ϕ.ker ≤ Q
```
(if I didn’t take a wrong turn).

This seems like it might only hold up to conjugation? So maybe I took a wrong turn in the proof somewhere?

Reply:
Try `P.1.ker_le_comap ϕ`?

Reply:
```
def comap_of_ker_is_p_group {p : ℕ} (P : sylow p G) {H : Type*} [group H] (ϕ : H →* G)
  (hϕ : is_p_group p ϕ.ker) (h : P.1 ≤ ϕ.range) : sylow p H :=
{ is_p_group' := P.2.comap_of_ker_is_p_group ϕ hϕ,
  is_maximal' := λ Q hQ hle, by
  { rw ← P.3 (hQ.map ϕ) (le_trans (ge_of_eq (map_comap_eq_self h)) (map_mono hle)),
    exact (comap_map_eq_self ((P.1.ker_le_comap ϕ).trans hle)).symm },
  .. P.1.comap ϕ }
```

Reply:
Thanks, I took this and put it into https://github.com/leanprover-community/mathlib/pull/11722 to keep PRs small